Migrate Executor/PrimaryChain to Operator/ConsensusChain

liuchengxu opened the pull request:
Pure refactoring (renamings), notable changes:

- `Executor` => `Operator`
- `ExecutorApi` => `DomainsApi`
- primary_foo, ExecutorFoo => consensus_foo, OperatorFoo
- crate _domain-executor_ => _domain-operator_
- crate _executor-gossip_ => _subnet-gossip_
- Log prefix is changed from `[PrimaryChain]` to `[Consensus]`

A majority of old usages are migrated to the new terms, but it's not guaranteed to eliminate all of them in this PR, you can still see them scattered in a bunch of places, please migrate them to the new terms when you see them in future development.

Close #1618

### Code contributor checklist:
* [x] I have read, understood and followed [contributing guide](https://github.com/subspace/subspace/blob/main/CONTRIBUTING.md)

liuchengxu committed:
Remove unused DomainConfig structure and unused deps in domain-executor

liuchengxu committed:
Rename ExecutorApi to DomainsApi

liuchengxu committed:
Update BundleHeader

liuchengxu committed:
Rename ExecutorSlotInfo to OperatorSlotInfo

liuchengxu committed:
Rename MockPrimaryNode to MockConsensusNode and update domain_runtime_code.rs and tests.rs in subspace-fraud-proof

- Rename `run_mock_primary_node()` to just `run()` as it's already under the MockConsensusNode context.
- Rename mock_primary_node to mock_consensus_node.

liuchengxu committed:
Rename crate domain-client-executor to domain-client-operator

_authors_ in Cargo.toml is changed to Subspace Labs as the current code
shares little similarities with the original fork.

liuchengxu committed:
Rename domain-client-executor-gossip to domain-client-subnet-gossip

liuchengxu committed:
Rename Executor keys to Operator keys

liuchengxu committed:
Update ExecutionReceipt

liuchengxu committed:
Rename EssentialExecutorParams to OperatorParams

liuchengxu committed:
Rename PBlock/PClient to CBlock/CClient in domain-executor

liuchengxu committed:
Rename Executor to Operator

Also renamed executor.rs to operator.rs

liuchengxu committed:
Tiny refactoring to consensus_block_info

Use HashAndNumber instead of a tuple.

liuchengxu committed:
Update subspace-node, subspace-service and domain-service

- primary_chain_config => consensus_chain_config
- primary_chain_node => consensus_chain_node
- ExecutorStreams => OperatorStreams
- PriamryChainTxPreValidator => ConsensusChainTxPreValidator

liuchengxu committed:
Update subspace-fraud-proof and block-preprocessor

liuchengxu committed:
Update primary things in domain_block_processor.rs and more in domain-executor

Another round of cleanup in domain-executor

Rename primary_block_hash to consensus_block_hash

nazar-pc reviewed (dismissed):
Some lines will exceed formatting length, but we have no strict rules about that, so not a big deal.

liuchengxu committed:
Update more primary and executor things

- domain-executor tests.
- runtime-configs
- executor_streams => operator_streams

nazar-pc reviewed (approved):
> I meant domain_block_number: consensus_block_number is only unused for the dummy receipts in tests

Only unused? I'm very confused as to why you're mentioning that something is unused, I had no idea that something is unused somewhere.

It is in test, so I guess it is fine if you expect block production on both sides to be the same, but you should seriously consider having at least a comment there because that assumption is very easy to break while code will continue to compile.
